ffi/capi/src/week.rs

```diff
@@ -81,6 +82,24 @@ pub mod ffi {
         pub fn min_week_days(&self) -> u8 {
             self.0.min_week_days
         }
+
+        #[diplomat::rust_link(icu::calendar::week::WeekCalculator::weekend, FnInStruct)]
+        pub fn weekend(&self) -> Box<ICU4XWeekCalculatorWeekendIterator> {
+            Box::new(ICU4XWeekCalculatorWeekendIterator(Box::new(
+                self.0.weekend(),
+            )))
+        }
+    }
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::calendar::week::WeekCalculator::weekend, FnInStruct)]
+    pub struct ICU4XWeekCalculatorWeekendIterator(Box<dyn Iterator<Item = IsoWeekday>>);
+
+    impl ICU4XWeekCalculatorWeekendIterator {
+        #[diplomat::rust_link(icu::calendar::week::WeekCalculator::weekend, FnInStruct)]
+        pub fn next(&mut self) -> Option<ICU4XIsoWeekday> {
+            self.0.next().map(|x| x.into())
+        }
     }
```

Comment:
Suggestion: do we want to transform the iterator into a bitset like we do with `ICU4XPluralCategories`?